api/client/workflows/doc_test.go

```diff
@@ -0,0 +1,127 @@
+package workflows_test
+
+// this package adds godoc examples for several Client types and functions
+// See https://pkg.go.dev/github.com/fluhus/godoc-tricks#Examples
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/workflows"
+)
+
+var ctx = context.Background()
+var plugin *workflows.Plugin
+var watcher workflows.RequestWatcher
+
+func ExampleNewPlugin() {
+	// Create a client with an open connection to a Teleport Auth server.
+	// documentation on createing a Teleport client can be found at
+	// pkg.go.dev/github.com/gravitational/teleport/api/client#New.
+	client, err := client.New(ctx, client.Config{
+		Addrs: []string{"proxy.example.com:3080"},
+		Credentials: []client.Credentials{
+			client.LoadIdentityFile("path/to/identity_file"),
+		},
+		InsecureAddressDiscovery: false,
+	})
+	if err != nil {
+		log.Fatalf("failed to create client: %v", err)
+
+	}
+	defer client.Close()
```

Comment:
Does the `plugin` need the `client` connection to live longer than this function?

Reply:
client needs stay up for plugin to function, I'll add a close function to plugin to make this possible in all scenarios.